src/components/inputs/text/NameTokenSelector/NameTokenSelector.tsx

```diff
@@ -0,0 +1,590 @@
+
+function NameTokenSelector({
+  walletAddress,
+  selectedTokenCallback,
+}: {
+  walletAddress: ArweaveTransactionID;
+  selectedTokenCallback: (id: ArweaveTransactionID | undefined) => void;
+}) {
+  const arweaveDataProvider = useArweaveCompositeProvider();
+  const [{ pdnsSourceContract }] = useGlobalState();
+
+  const [searchText, setSearchText] = useState<string>();
+  const [tokens, setTokens] = useState<{
+    [x: string]: Partial<PDNTContractJSON>;
+  }>();
+  const [loading, setLoading] = useState(false);
+  const [filteredTokens, setFilteredTokens] =
+    useState<
+      Array<{ id: string; name?: string; ticker?: string } | undefined>
+    >();
+  const [selectedToken, setSelectedToken] = useState<
+    { id: string; name: string; ticker: string } | undefined
+  >(undefined);
+  const [searching, setSearching] = useState<boolean>(false);
+  const [searchActive, setSearchActive] = useState(false);
+  const [validImport, setValidImport] = useState<boolean | undefined>(
+    undefined,
+  );
+
+  const listRef = useRef<HTMLDivElement>(null);
+  const [listPage, setListPage] = useState<number>(1);
+  const listItemCount = 5;
+
+  useEffect(() => {
+    selectedTokenCallback(
+      selectedToken ? new ArweaveTransactionID(selectedToken.id) : undefined,
+    );
+  }, [selectedToken]);
+
+  useEffect(() => {
+    getTokenList(walletAddress);
+  }, [walletAddress]);
+
+  useEffect(() => {
+    // token import update
+    if (searchText) {
+      handleTokenSearch(searchText);
+      setListPage(1);
+    }
+  }, [tokens]);
+
+  useEffect(() => {
+    if (!listRef.current) {
+      return;
+    }
+    document.addEventListener('mousedown', handleClickOutside);
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [listRef]);
+
+  function handleClickOutside(e: any) {
+    e.preventDefault();
+    if (
+      listRef.current &&
+      e.target !== listRef.current &&
+      !listRef.current.contains(e.target)
+    ) {
+      setSearchText('');
+      setSearchActive(false);
+      setFilteredTokens(undefined);
+      setListPage(1);
+    }
+  }
+
+  async function getTokenList(
+    address: ArweaveTransactionID,
+    imports?: ArweaveTransactionID[],
+  ) {
+    try {
+      setLoading(true);
+      const fetchedIds = await arweaveDataProvider.getContractsForWallet(
+        pdnsSourceContract.approvedANTSourceCodeTxs.map(
+          (sourceCode) => new ArweaveTransactionID(sourceCode),
+        ),
+        address,
+      );
+      if (!fetchedIds.ids) {
+        throw new Error(
+          'Unable to find any Name Tokens for the provided address',
+        );
+      }
+      if (imports) {
+        imports.map((id) =>
+          arweaveDataProvider
+            .validateTransactionTags({
+              id: id.toString(),
+              requiredTags: {
+                'Contract-Src': pdnsSourceContract.approvedANTSourceCodeTxs,
+              },
+            })
```

Comment:
we've decided we'll no longer be validating if an ANT contract is in an approved list - so we can likely remove this check.

Reply:
switched this to simply checking if its a smartweave contract for now. Thats about the minimum validation we should do here.